lesson-7/server/api.js

```diff
@@ -0,0 +1,56 @@
+const MongoClient = require('mongodb').MongoClient;
+const config = require('./config');
+
+function executeQuery(query) {
+    return new Promise((resolve, reject) => {
+        MongoClient.connect(config.MONGO_URL).then(database => {
+            var todos = query(database)
+                .then(result => {
+                    resolve(result);
+                })
+                .catch(err => {
+                    reject('query failed: ' + err);
+                })
+        }).catch(err => {
+            reject('connection failed: ' + err);
+        });
+    });  
+}
+
+function getTodoItems() {
+    return executeQuery(database => {
```

Comment:
`executeQuery` returns Promise but here it is passed callback?

Reply:
used conneciton pooling instead https://mongodb.github.io/node-mongodb-native/driver-articles/mongoclient.html